v4/internal/otel_expect.go

```diff
@@ -0,0 +1,40 @@
+package internal
```

Comment:
This file is where the birthday cake ice cream lies.

The `OpenTelemetryExpect` in this file wraps a `*testtrace.Tracer` and adds all the methods required for `internal.Expect`. This way we can easily access the data saved to the Tracer when testing. In the future when we add otel metrics to the shim, we can simply imbed the otel metric type here as well.